### PR TITLE
🐛Use generate name for CP loadbalancer listeners

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Currently, we use timestamps to generate names for target rules of additional CP load balancer listeners. This fails because the name is generated so quickly that the timestamp stays the same. In this PR I changed timestamps to `GenerateName` from the API server, this should generate a unique and valid name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Use generate name for CP loadbalancer listeners
```

### CODE REVIEW

**Comment by alexander-demicev** [comment]
@nrb E2E failures don't seem to be related to the PR

**Comment by nrb** [comment]
Yeah, looks like Boskos errors. Will re-run while waiting for a second review.

/retest
/approve
/hold
/assign @damdo

Feel free to unhold once you've reviewed @damdo

File: pkg/cloud/services/elb/loadbalancer.go [hunk 1]
```diff
@@ -288,7 +289,7 @@ func (s *Service) getAPIServerLBSpec(elbName string, lbSpec *infrav1.AWSLoadBala
 				Protocol: listener.Protocol,
 				Port:     listener.Port,
 				TargetGroup: infrav1.TargetGroupSpec{
-					Name:        fmt.Sprintf("additional-listener-%d", time.Now().Unix()),
+					Name:        names.SimpleNameGenerator.GenerateName("apiserver-target-"),
```
Comment (damdo) [inline pkg/cloud/services/elb/loadbalancer.go:292]: Should this be `additional -listener` ?

**Review by damdo — COMMENTED** [review]
Thanks for this Alex :)

File: pkg/cloud/services/elb/loadbalancer.go (hunk 1 above)
Comment (alexander-demicev) [reply pkg/cloud/services/elb/loadbalancer.go]: yes, it should, thanks

**Review by damdo — COMMENTED** [review]
/lgtm

**Pull request merged**